src/ast/mod.rs

```diff
@@ -7766,6 +7766,27 @@ impl fmt::Display for RenameTable {
     }
 }
 
+/// table object for insertion
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub enum TableObject {
+    // for simple table name
+    TableName(#[cfg_attr(feature = "visitor", visit(with = "visit_relation"))] ObjectName),
+
+    // for Clickhouse [table functions](https://clickhouse.com/docs/en/sql-reference/table-functions)
+    TableFunction(Function),
+}
+
+impl fmt::Display for TableObject {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::TableName(table_name) => write!(f, "{}", display_separated(&table_name.0, ".")),
```

Comment:
```suggestion
            Self::TableName(table_name) => write!(f, "{table_name}"),
```
I think this something like this should be equivalent?